source/git/gitsource_unix.go

```diff
@@ -4,26 +4,83 @@ package git
 
 import (
 	"context"
+	"os"
 	"os/exec"
-	"syscall"
+	"os/signal"
 	"time"
+
+	"github.com/docker/docker/pkg/reexec"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	gitCmd = "umask-git"
 )
 
+func init() {
+	reexec.Register(gitCmd, gitMain)
+}
+
+func gitMain() {
+	// Need standard user umask for git process.
+	unix.Umask(0022)
+
+	// Reexec git command
+	cmd := exec.Command(os.Args[1], os.Args[2:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Stdin = os.Stdin
+
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan,
+		unix.SIGCONT,
```

Comment:
do we need to define individual signals?

Reply:
I catch all signal through `moby/sys` if it's ok.

Reply:
I meant what about all signals. For the current, it seems to be broken compared to the old code that sends the signal to whole group.

Reply:
Done